Tasks/AzureFileCopyV2/AzureFileCopy.ps1

```diff
@@ -0,0 +1,250 @@
+    {
+        $vmsAdminUserName = ".\" + $vmsAdminUserName 
+    }
+    # getting azure vms properties(name, fqdn, winrmhttps port)
+    $azureVMResourcesProperties = Get-AzureVMResourcesProperties -resourceGroupName $environmentName -connectionType $connectionType `
+    -resourceFilteringMethod $resourceFilteringMethod -machineNames $machineNames -enableCopyPrerequisites $enableCopyPrerequisites -connectedServiceName $connectedServiceName
+
+    $azureVMsCredentials = Get-AzureVMsCredentials -vmsAdminUserName $vmsAdminUserName -vmsAdminPassword $vmsAdminPassword
+
+    # Get Invoke-RemoteScript parameters
+    $invokeRemoteScriptParams = Get-InvokeRemoteScriptParameters -azureVMResourcesProperties $azureVMResourcesProperties `
+                                                                 -networkCredentials $azureVMsCredentials `
+                                                                 -skipCACheck $skipCACheck
+
+    # generate container sas token with full permissions
+    $containerSasToken = Generate-AzureStorageContainerSASToken -containerName $containerName -storageContext $storageContext -tokenTimeOutInHours $defaultSasTokenTimeOutInHours
+
+    # Copies files on azureVMs 
+    Copy-FilesToAzureVMsFromStorageContainer -targetMachineNames $invokeRemoteScriptParams.targetMachineNames `
+                                             -credential $invokeRemoteScriptParams.credential `
+                                             -protocol $invokeRemoteScriptParams.protocol `
+                                             -sessionOption $invokeRemoteScriptParams.sessionOption `
+                                             -blobStorageEndpoint $blobStorageEndpoint `
+                                             -containerName $containerName `
+                                             -containerSasToken $containerSasToken `
+                                             -targetPath $targetPath `
+                                             -cleanTargetBeforeCopy $cleanTargetBeforeCopy `
+                                             -copyFilesInParallel $copyFilesInParallel `
+                                             -additionalArguments $additionalArgumentsForVMCopy `
+                                             -azCopyToolLocation $azCopyLocation `
+                                             -fileCopyJobScript $AzureFileCopyRemoteJob `
+                                             -enableDetailedLogging $enableDetailedLogging
+
+    Write-Output (Get-VstsLocString -Key "AFC_CopySuccessful" -ArgumentList $sourcePath, $environmentName)
+}
+catch
+{
+    Write-Verbose $_.Exception.ToString()
+
+    Write-Telemetry "Task_InternalError" "CopyingToAzureVMFailed"
+    throw
+}
+finally
+{
+    Remove-AzureContainer -containerName $containerName -storageContext $storageContext
```

Comment:
does Remove-AzureContainer this will get executed in blob upload flow as well, may be I am incorrectly reading the flow

Reply:
this try-catch-finally will only be executed for VM destination.